msal/src/main/java/com/microsoft/identity/client/controllers/LocalMSALController.java

```diff
@@ -134,26 +140,138 @@ private TokenResult performTokenRequest(OAuth2Strategy strategy, AuthorizationRe
     }
 
-    void throwIfNetworkNotAvailable(Context context) throws MsalClientException {
+    void throwIfNetworkNotAvailable(final Context context) throws MsalClientException {
         final ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         final NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+
         if (networkInfo == null || !networkInfo.isConnected()) {
-            throw new MsalClientException(MsalClientException.DEVICE_NETWORK_NOT_AVAILABLE, "Device network connection is not available.");
+            throw new MsalClientException(
+                    MsalClientException.DEVICE_NETWORK_NOT_AVAILABLE,
+                    "Device network connection is not available."
+            );
         }
     }
 
-    private ICacheRecord saveTokens(OAuth2Strategy strategy, AuthorizationRequest request, TokenResponse tokenResponse, OAuth2TokenCache tokenCache) throws ClientException {
+    private ICacheRecord saveTokens(final OAuth2Strategy strategy,
+                                    final AuthorizationRequest request,
+                                    final TokenResponse tokenResponse,
+                                    final OAuth2TokenCache tokenCache) throws ClientException {
         return tokenCache.save(strategy, request, tokenResponse);
     }
 
-
     @Override
-    public void completeAcquireToken(int requestCode, int resultCode, final Intent data) {
+    public void completeAcquireToken(final int requestCode,
+                                     final int resultCode,
+                                     final Intent data) {
         mAuthorizationStrategy.completeAuthorization(requestCode, resultCode, data);
     }
 
     @Override
-    public AcquireTokenResult acquireTokenSilent(MSALAcquireTokenSilentOperationParameters request) {
-        throw new UnsupportedOperationException();
+    public AcquireTokenResult acquireTokenSilent(
+            final MSALAcquireTokenSilentOperationParameters parameters)
+            throws MsalClientException, IOException, ClientException {
+        final AcquireTokenResult acquireTokenSilentResult = new AcquireTokenResult();
+        final OAuth2TokenCache tokenCache = parameters.getTokenCache();
+
+        final String environment = parameters.getAuthority().getAuthorityURL().getHost();
+        final String clientId = parameters.getClientId();
+        final String homeAccountId =
+                parameters
+                        .getAccount()
+                        .getHomeAccountIdentifier()
+                        .getIdentifier();
+
+        final Account targetAccount = tokenCache.getAccount(
+                environment,
+                clientId,
+                homeAccountId
+        );
+
+        final OAuth2Strategy strategy = parameters.getAuthority().createOAuth2Strategy();
+        final ICacheRecord cacheRecord = tokenCache.load(clientId, targetAccount);
```

Comment:
I think i'm missing something here.  Where did you say which access token that you're looking for?  The account could have multiple access tokens?